project_generator/tools/iar.py

```diff
@@ -138,13 +138,18 @@ def _ewp_set_name(self, ewp_dic, name):
     def _eww_set_path_single_project(self, eww_dic, name):
         eww_dic['workspace']['project']['path'] = join('$WS_DIR$', name + '.ewp')
 
-    def _eww_set_path_multiple_project(self, eww_dic, projects):
+    def _eww_set_path_multiple_project(self, eww_dic):
         eww_dic['workspace']['project'] = []
         for project in self.workspace['projects']:
-            # This has an assumption all projects inside workspace
-            path_to_project = project['output_dir']['path']
-            path_to_project = path_to_project.replace(self.workspace['settings']['path'] + '\\', '', 1)
-            eww_dic['workspace']['project'].append( { 'path' : join('$WS_DIR$', path_to_project, project['name'] + '.ewp') })
+            # We check how far is project from root and workspace. IF they dont match,
+            # get relpath for project and inject it into workspace
+            path_project = os.path.dirname(project['output_dir']['path'] + '\\')
+            path_workspace = os.path.dirname(self.workspace['settings']['path'] + '\\')
+            # path_to_project = path_to_project.replace( + '\\', '', 1)
+            if path_project != path_workspace:
+                rel_path = os.path.relpath(os.getcwd(), path_workspace)
+                path_project = os.path.join(rel_path, path_project)
+            eww_dic['workspace']['project'].append( { 'path' : join('$WS_DIR$', path_project, project['name'] + '.ewp') })
```

Comment:
all the code above should be somewhere in project.py . Tool should get paths for workspace ready. As we can see this is the same for uvision and iar or any other tool. It's just get where is workspace generated, where is a project. If they differ, get some logic to get propre project path from a workspace.

@sarahmarshy You might be interested in this